tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -747,5 +747,29 @@ public void KillEverything ()
 			// https://github.com/xamarin/xamarin-macios/issues/10012
 			ExecutionHelper.Execute ("xcrun", new [] { "simctl", "list" });
 		}
+
+
+		[TestCase (ApplePlatform.iOS, "MyShareExtension.csproj", "bin/Debug/net6.0-ios/iossimulator-x64/MySimpleApp.app/PlugIns/MyShareExtension.appex")]
+		[TestCase (ApplePlatform.TVOS, "MyTVServicesExtension.csproj", "bin/Debug/net6.0-tvos/tvossimulator-x64/MySimpleApp.app/PlugIns/MyTVServicesExtension.appex")]		
+		[TestCase (ApplePlatform.MacOSX, "ShareExtensionTest.csproj", "bin/Debug/net6.0-macos/osx-x64/MySimpleApp.app/Contents/PlugIns/ShareExtensionTest.appex")]
+		// [TestCase ("MacCatalyst", "")] - No extension support yet
+		public void BuildProjectsWithExtensions (ApplePlatform platform, string extensionProjectName, string appPath)
+		{
+			Configuration.IgnoreIfIgnoredPlatform (platform);
+			var consumingProjectDir = GetProjectPath ("ExtensionConsumer", platform: platform);
+			var extensionProjectDir = GetProjectPath ("ExtensionProject", platform: platform);
+
+			Clean (extensionProjectDir);
+			Clean (consumingProjectDir);
+
+			Configuration.CopyDotNetSupportingFiles (Path.GetDirectoryName(extensionProjectDir));
+			Configuration.CopyDotNetSupportingFiles (Path.GetDirectoryName(consumingProjectDir));
+
+			DotNet.AssertBuild (extensionProjectDir, verbosity);
```

Comment:
You shouldn't have to build the extension project manually first, just building the consuming project should be enough.